test/src/test/bash/test_zstack_server_mariadb_recovery.sh

```
@@ -0,0 +1,102 @@
+PATH="$bin_dir:$PATH" \
+ZSTUB_CALLS="$calls" \
+ZSTUB_MARIADB_ACTIVE=true \
+ZSTACK_MARIADB_SOCKET="$socket" \
+ZSTACK_SERVER_LOG_DIR="$log_dir" \
+sh "$script" start >/dev/null
+
+[ -e "$socket" ] || {
+    echo "expected active MariaDB socket to be kept" >&2
+    exit 1
+}
+! grep -F "systemctl start mariadb" "$calls" >/dev/null
```

Shellcheck (0.11.0)

[info] This ! is not on a condition and skips errexit. Use `&& exit 1` instead, or make sure $? is checked.

(SC2251)